index.js

```diff
@@ -94,12 +94,12 @@ exports.init = function (sbot, config) {
   }
 
   sbot.on('rpc:connect', function (rpc, isClient) {
-    if (isClient) {
+    if (isClient && rpc.id !== sbot.id) {
```

Comment:
Can you add a comment to this. If I hadn't read the PR description I wouldn't have guessed what the problem was. Maybe do that one a line first like: `if (rpc.id === sbot.id) return // ssb-client connection will trigger this`.

I'm guessing you have something similar in ssb-conn for these rpc:connect cases? Seems like quite a gotcha.

Reply:
Yes ssb-conn-hub has to treat this, https://github.com/staltz/ssb-conn-hub/blob/f4e66db2e256cf2b54ff814a49cd7759ee0f2c16/src/index.ts#L119

It's a gotcha, and overall I'd like to reduce our usage of Node.js Emitter events in the SSB stack, including `rpc:connect`. With ssb-db2 migrate progress events, we discovered some major performance headaches there, and I'm thinking that we could just use a pull-notify instead.